pkg/platform/kube/functionres/lazy.go

```diff
@@ -2500,6 +2504,8 @@ func (lc *lazyClient) populateDeploymentContainer(ctx context.Context,
 		FailureThreshold:    function.Spec.ReadinessProbe.FailureThreshold,
 	}
 
+	// enrichment ensures backward compatibility for functions created with controller versions < v1.14.5, where probes may be nil
+	common.EnrichProbe(&function.Spec.LivenessProbe, defaultPlatformConfiguration.Kube.DefaultLivenessProbe)
```

Comment:
avoid duplicating the comment, and do the enrichment for both probes one after the other (meaning, move this to line 2494)